controllers/profiles/reconciler_prod_test.go

```diff
@@ -129,3 +130,34 @@ func Test_deployWorkflowReconciliationHandler_handleObjects(t *testing.T) {
 	*/
 
 }
+
+func Test_GenerationAnnotationCheck(t *testing.T) {
+	logger := ctrllog.FromContext(context.TODO())
+	// we load a workflow with metadata.generation to 0
+	workflow := test.GetKogitoServerlessWorkflow("../../config/samples/"+test.KogitoServerlessWorkflowSampleYamlCR, t.Name())
+	// make sure that the workflow won't trigger a change
+	workflow.Status.Applied = workflow.Spec
+	platform := test.GetKogitoServerlessPlatformInReadyPhase("../../config/samples/"+test.KogitoServerlessPlatformWithCacheYamlCR, t.Name())
+	client := test.NewKogitoClientBuilder().WithRuntimeObjects(workflow, platform).Build()
+	handler := &deployWorkflowReconciliationState{
+		stateSupport: fakeReconcilerSupport(client),
+		ensurers:     newProdObjectEnsurers(&stateSupport{logger: &logger, client: client}),
+	}
+	result, objects, err := handler.Do(context.TODO(), workflow)
+	assert.Greater(t, result.RequeueAfter, int64(0))
+	assert.NoError(t, err)
+	assert.NotNil(t, result)
+	assert.Len(t, objects, 2)
+	// then we load a workflow with metadata.generation set to 1
```

Comment:
instead update the generation, imho, would be better to compare if the content of the workflow has changed.

Reply:
I can add the compare the content of the workflow, but is the metadata.generation value compared to the status.observedGeneration the kickstart for the new build in this new code